examples/move/scratch_off/sources/game.move

```diff
@@ -0,0 +1,646 @@
+    }
+
+    public fun leaderboard(store: &ConvenienceStore): &LeaderBoard {
+       &store.leaderboard 
+    }
+
+    public fun leaderboard_players(leaderboard: &LeaderBoard): vector<address> {
+        leaderboard.leaderboard_players
+    }
+
+    public fun winning_tickets(store: &ConvenienceStore): &vector<PrizeStruct> {
+        &store.winning_tickets
+    }
+
+    public fun table_contains_player(player_metadata: &Table<address, Metadata>, target_address: address): bool {
+        table::contains(player_metadata, target_address)
+    }
+
+    public fun player_metadata(store: &ConvenienceStore): &Table<address, Metadata> {
+        &store.player_metadata
+    }
+
+    public fun get_target_player_metadata(player_metadata: &Table<address, Metadata>, target_address: address): &Metadata {
+        table::borrow(player_metadata, target_address)
+    }
+
+    public fun get_player_metadata_mut(player_metadata: &mut Table<address, Metadata>, target_address: address): &mut Metadata {
+        table::borrow_mut(player_metadata, target_address)
+    }
+
+    public fun original_ticket_count(store: &ConvenienceStore): u64 {
+        store.original_ticket_count
+    }
+
+    public fun tickets_left(store: &ConvenienceStore): u64 {
+        store.original_ticket_count - store.tickets_evaluated
+    }
+
+    public fun tickets_issued(store: &ConvenienceStore): u64 {
+        store.tickets_issued
+    }
+
+    public fun prize_pool_balance(store: &ConvenienceStore): u64 {
+        balance::value(&store.prize_pool)
+    }
+
+    // --------------- House Accessors ---------------
+
+    public fun public_key(store: &ConvenienceStore): vector<u8> {
+        store.public_key
+    }
+
+    public fun ticket_exists(store: &ConvenienceStore, ticket_id: ID): bool {
+        dof::exists_with_type<ID, Ticket>(&store.id, ticket_id)
+    }
+
+    // Tests
+    #[test_only]
+    public fun init_for_testing(ctx: &mut TxContext){
+        let store_uid = object::new(ctx);
+        let new_store = ConvenienceStore {
+            id: store_uid,
+            creator: tx_context::sender(ctx),
+            prize_pool: balance::zero(),
+            winning_tickets: vector[], 
+            original_ticket_count: 0,
+            tickets_issued: 0, 
+            tickets_evaluated: 0,
+            player_metadata: table::new(ctx),
+            leaderboard: LeaderBoard {
+                lowest_sui_won: 0,
+                max_players: 10,
+                leaderboard_players: vector[],
+                leaderboard_player_metadata: table::new(ctx),
+            },
+            public_key: vector[]
+        };
+        transfer::share_object(new_store);
+        transfer::public_transfer(StoreCap {
+            id: object::new(ctx),
+        }, tx_context::sender(ctx));
+        transfer::public_transfer(SetUpCap {
+            id: object::new(ctx),
+        }, tx_context::sender(ctx));
+    }
+    
+    #[test_only]
+    public fun finish_evaluation_for_testing(
```

Comment:
You should generally create an internal function if you wanna re-use almost all the logic apart from one validation for testing (It's way easier to avoid mistakes or having different code in one + different in the other)